OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -105,6 +105,11 @@ public static void ProcessOrder(OrderManager orderManager, World world, int clie
 							break;
 						}
 
+						// Save most recently chosen Faction.
+						var client = orderManager.LobbyInfo.ClientWithIndex(clientId);
+						Game.Settings.Game.LastFaction[Game.ModData.Manifest.Id] = client.Faction;
```

Comment:
This looks like the wrong place to be doing this.  You probably want to do this from the lobby code instead, and will also need to be careful about the preferred / server state (see how we handle player colors).